### PR TITLE
Fixed a number of warnings

### PR DESCRIPTION
**Pull request by oscargus**
Got rid of a number of warnings:

 - <>
 - static methods
 - commented empty blocks
 - general clean up

### CODE REVIEW

File: src/main/java/net/sf/jabref/JabRef.java [hunk 1]
```diff
@@ -160,7 +160,7 @@ public void start(String[] args) {
         openWindow(loaded);
     }
 
-    private void setupLogHandlerForErrorConsole() {
+    private static void setupLogHandlerForErrorConsole() {
```
Comment (lenhard) [inline src/main/java/net/sf/jabref/JabRef.java]: Why do you turn this static? It is not called from a static context.

**Comment by lenhard** [comment]
I find this PR very valuable (we need such improvements). The only thing I am unsure about is turning various methods static (see above). Can you explain the rationale behind this? Why is static needed here?

**Comment by koppor** [comment]
The `static` conversion is because Eclipse warns if methods could be made static. On the one hand, we could add `@suppresswarning` if we are sure that Eclipse is wrong. On the other hand, we could just turn off the `static` warning. I don't have strong a opinion about that. I currently lean towards making methods static if possible to show that they don't access instance variables.

**Comment by oscargus** [comment]
As @koppor correctly pointed out, this is based on Eclipse warnings and
Eclipse warnings only. I've trusted it so far, but recently I've noted that
some warnings disappear when I fix other, to me seemingly unrelated,
warnings, so maybe I should read up a bit...

**Comment by lenhard** [comment]
I tend to disagree with the warning and would prefer non-static versions. The problem with the use of static is that it leads to a higher degree of coupling within the system. Every user of a static method uses the very same instance (that is, the class) and this transitively couples two users although there might be no need for it. 

Don't get me wrong, the code works with both versions. Especially if the class is instantiated only once, as is the case in the above line I commented at, there is no difference execution-wise. It is merely a "design smell". I have the opinion that a design with a lot of use of static is "worse" than a more object-oriented design. If we consistently favor static over non-static, we end up with a purely imperative program, which means we are propably using the wrong programming language.

@JabRef/developers This is about a coding guideline and concerns everyone. So what are the opinions of the remaining people? Should we consistently fix the Eclipse warning regarding the use of static, or should we turn it off?

**Comment by oscargus** [comment]
OK, I've read up and see the point. I'd say that maybe one should go through all the static methods and remove the static keyword and add warning removal in case it does not make sense to access the method in a static way.

For reference: I've done this in the earlier cleanups as well, so it may be better to go over it once and for all (although I can of course start by changing these ones back).

**Comment by matthiasgeiger** [comment]
I agree with @lenhard - we should try to reduce the usage of `static` methods and not introduce new ones. 
Quick rule of thumb: 
> Static is sensible in Utility classes which provide independent small methods, e.g, `getIntValue(...)`. Static should be avoided in complexer "logic classes" - even if only static or no members are accessed. 

Objections? @lenhard 

I don't like `@suppressWarnings` annotations - so would either turn off this warning completely - or just ignore those warnings "manually", i.e., think about it and decide whether the usage is okay or not.

**Comment by lenhard** [comment]
@matthiasgeiger full ack :+1:

**Comment by koppor** [comment]
Could you review eb441846817b11 and efce4ec2eae5bd79e30 (unrelated, but I think, it's important, too)

**Comment by lenhard** [comment]
@koppor Looks got to me. I imported the project into Eclipse and am not getting the warning at the above position. I also agree with you on the usage of "this": it should not be mandatory.

However, I found another Eclipse warning, we should be aware of here. In the following line
``` java
new FocusRequester(((BasePanel) JabRef.jrf.tabbedPane.getComponentAt(0)).mainTable);
```
Eclipse warns that `The allocated object is never used`. The warning is a false positive. `FocusRequester` starts itself in a separate Thread in its constructor (which is unsafe publication and a potential bug, but that is a different issue) and it does not matter that it is not used at this position. I don't think we should ignore the `unused` warning, but *we should not blindly delete every variable for which it occurs*. Instead, we should add a `@SuppressWarnings("unused")`, if it is a false positive.

**Comment by oscargus** [comment]
I have removed the statics introduced in this PR.

**Comment by oscargus** [comment]
Rebased on current master so it should be possible to merge this (unless any other issues arise).

File: src/main/java/net/sf/jabref/bst/VM.java [hunk 2]
```diff
@@ -1185,17 +1185,16 @@ private void iterate(Tree child) {
     }
 
     /**
-     * Sorts the entry list using the values of the string entry variable
-     * sort.key$. It has no arguments.
-     * 
-     * @param child
+     * Sorts the entry list using the values of the string entry variable sort.key$. It has no arguments.
+     *
+     * @param
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/bst/VM.java]: `@param` should be removed here completely

File: src/main/java/net/sf/jabref/exporter/layout/LayoutHelper.java [hunk 3]
```diff
@@ -342,26 +283,22 @@ private Object parse() throws IOException, StringIndexOutOfBoundsException {
     }
 
     /**
-     * 
+     *
      */
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/exporter/layout/LayoutHelper.java]: `/** */` block can be removed completely

File: src/main/java/net/sf/jabref/exporter/layout/LayoutHelper.java [hunk 4]
```diff
@@ -371,83 +308,64 @@ private void parseField() throws IOException, StringIndexOutOfBoundsException
                     firstLetter = name.charAt(0);
                 } catch (StringIndexOutOfBoundsException ex) {
                     StringBuilder lastFive = new StringBuilder(10);
-                    for (StringInt entry : parsedEntries.subList(Math.max(0, parsedEntries.size() - 6), parsedEntries.size() - 1)) {
+                    for (StringInt entry : parsedEntries.subList(Math.max(0, parsedEntries.size() - 6),
+                            parsedEntries.size() - 1)) {
                         lastFive.append(entry.s);
                     }
-                    throw new StringIndexOutOfBoundsException(Localization.lang("Backslash parsing error near") + " \'" + lastFive.toString().replace("\n", " ") + '\'');
+                    throw new StringIndexOutOfBoundsException(Localization.lang("Backslash parsing error near") + " \'"
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/exporter/layout/LayoutHelper.java]: To be checked: Is localization needed here? Should be only relevant if the exception message is shown somewhere in the UI.

File: src/main/java/net/sf/jabref/groups/AutoGroupDialog.java [hunk 5]
```diff
@@ -96,7 +96,7 @@ public void actionPerformed(ActionEvent e) {
                         new ExplicitGroup(Localization.lang("Automatically created groups"),
                                 GroupHierarchyType.INCLUDING));
                 Set<String> hs = null;
-                String field = field();
+                String field1 = field();
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/groups/AutoGroupDialog.java]: Maybe renaming the variable to "fieldText" would be more intuitive. 

Side note: Why the hell has the method "field()" been introduced, which does nothing more than returning `field.getText()`? 

@oscargus Would you remove those rather useless methods (`field(), remove()`), too? Thanks! :wink:

File: src/main/java/net/sf/jabref/gui/BasePanel.java [hunk 6]
```diff
@@ -495,8 +482,7 @@ private void setupActions() {
         // The action for opening the string editor
         actions.put(Actions.EDIT_STRINGS, (BaseAction) () -> {
             if (stringDialog == null) {
-                StringDialog form = new StringDialog
-                        (frame, BasePanel.this, database, Globals.prefs);
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/gui/BasePanel.java]: Those were really weired line breaks... Good to see that fixed :+1:

File: src/main/java/net/sf/jabref/importer/fileformat/InspecImporter.java [hunk 7]
```diff
@@ -82,26 +80,25 @@ public boolean isRecognizedFormat(InputStream stream)
...
         String str;
-        while ((str = in.readLine()) != null) {
-            if (str.length() < 2) {
-                continue;
-            }
-            if (str.indexOf("Record") == 0) {
-                sb.append("__::__").append(str);
-            } else {
-                sb.append("__NEWFIELD__").append(str);
+        try (BufferedReader in = new BufferedReader(ImportFormatReader.getReaderDefaultEncoding(stream))) {
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/importer/fileformat/InspecImporter.java]: :+1:

File: src/main/java/net/sf/jabref/wizard/auximport/AuxSubGenerator.java [hunk 8]
```diff
@@ -224,8 +222,8 @@ private boolean parseAuxFile(String filename) {
                             }
                         }
                     }
-                } // line != null
-                else {
+                } else {
+                    // line != null
                     weiter = false;
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/wizard/auximport/AuxSubGenerator.java]: "weiter" should be renamed to "continue" for people who don't speak German :wink:

File: src/main/java/net/sf/jabref/wizard/auximport/AuxSubGenerator.java [hunk 9]
```diff
@@ -300,12 +299,11 @@ private void resolveTags() {
      * @param auxDB The database to insert into.
```
Comment (matthiasgeiger) [inline src/main/java/net/sf/jabref/wizard/auximport/AuxSubGenerator.java]: param must be renamed to

**Comment by matthiasgeiger** [comment]
Huh.... requires quite some time to scroll through all your changes :wink:

As you can see above I had a few remarks. 

Apart from those: Good job! Should be ready to merge (which should be easy after your rebasing - bonus kudos for this :trophy:)

File: src/main/java/net/sf/jabref/exporter/layout/LayoutHelper.java (hunk 4 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/exporter/layout/LayoutHelper.java]: No. However, there are some localized exception and log messages in the current code and these will be removed in one go and the translation files updated at the same time.

File: src/main/java/net/sf/jabref/wizard/auximport/AuxSubGenerator.java (hunk 8 above)
Comment (oscargus) [inline src/main/java/net/sf/jabref/wizard/auximport/AuxSubGenerator.java]: I do not speak enough German to be fully sure, but that was my interpretation of it as well. :-) (Except that continue is a quite bad choice... ;-))

**Comment by oscargus** [comment]
Thanks for the review! Should be good to go now.

**Pull request merged**